### ISSUE TITLE
Cannot infer 'go.mod' or 'go.work' from parents

### ISSUE THREAD

**Issue by joaohaas** [golang/vscode-go#2267]
**Is your feature request related to a problem? Please describe.**
I have a monorepo for a backend where the structure looks something like this:

```
- root
| - golang
| | - a
| | | - go.mod
| | | - main.go
| | - b
| | | - go.mod
| | | - main.go
| | - ...
| - ...
```

Whenever I open VSCode on the `root` folder, the `.go` files can't understand the 'module context' they're supposed to be in. For example, if `golang/a/go.mod` has `require github.com/lib/pq`, and `golang/a/main.go` tries to `import "github.com/lib/pq"`, an error will be thrown.

**Describe the solution you'd like**
Ideally, when editing `golang/a/**/*.go`, each file should have a 'module context' of `golang/a/go.mod`, or the first `go.mod` found on the parents. The same result should be expected for `a`, `b`, etc.
Ideally this would happen automatically, but I think creating a `go.work` inside the `golang` folder would also be a valid solution.

**Describe alternatives you've considered**
Opening each project on separate windows works, but it really hinders my workflow. Creating a `go.work` with `use ./a`, `use ./b`, etc, on the `golang` folder works if I open VSCode on the `golang` folder, but not when I open on the `root` folder.

**Additional context**
The error that appears on `golang/a/main.go` when I try to use a dependency required on `golang/a/go.mod`, in this case, `github.com/lib/pq`:
```
could not import github.com/lib/pq (cannot find package "github.com/lib/pq" in any of 
	/usr/local/go/src/github.com/lib/pq (from $GOROOT)
	/Users/<username>/go/src/github.com/lib/pq (from $GOPATH))compiler[BrokenImport](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal?utm_source%3Dgopls#BrokenImport)
```

The error does not appear when opening the folder directly, since go modules work correctly in that case.

**Comment by findleyr** [comment]
This is a dupe of #2249 (interesting that two such similar issues arrived recently...)

See my comment here: https://github.com/golang/vscode-go/issues/2249#issuecomment-1142158815

TL;DR it is not obvious what gopls should do when there are multiple nested modules, and furthermore we want to produce identical behavior as the Go command. `go.work` is the way to solve these problems.
